rest/api_test.go

```diff
@@ -4466,3 +4467,205 @@ func Benchmark_RestApiGetDocPerformanceFullRevCache(b *testing.B) {
 		}
 	})
 }
+
+func TestHandleProfiling(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	dirPath := fmt.Sprintf("%vpprof", os.TempDir())
+	assert.NoError(t, os.MkdirAll(dirPath, 0755))
+	defer func() { assert.NoError(t, os.RemoveAll(dirPath)) }()
+
+	tests := []struct {
+		inputProfile string
+	}{
+		{inputProfile: "goroutine"},
+		{inputProfile: "threadcreate"},
+		{inputProfile: "heap"},
+		{inputProfile: "allocs"},
+		{inputProfile: "block"},
+		{inputProfile: "mutex"},
+	}
+
+	for _, tc := range tests {
+		// Send a valid profile request.
+		resource := fmt.Sprintf("/_profile/%v", tc.inputProfile)
```

Comment:
All of these tests calling the profiling endpoints take a decent chunk of time each - maybe we should skip these when running in short mode, or evaluate whether we actually need to test these at all.

Reply:
Yeah, it takes close to 12 seconds during each run; let's skip these when running in short mode.